src/main/java/build/buildfarm/server/AdminService.java

```diff
@@ -63,6 +65,23 @@ public void stopContainer(StopContainerRequest request, StreamObserver<Status> r
     }
   }
 
+  @Override
+  public void getHosts(GetHostsRequest request, StreamObserver<GetHostsResult> responseObserver) {
+    try {
+      GetHostsResult result = null;
+      if (adminController != null) {
+        result =
+            adminController.getHosts(
+                request.getFilter(), request.getAgeInMinutes(), request.getStatus());
+      }
+      responseObserver.onNext(result);
+      responseObserver.onCompleted();
+    } catch (Exception e) {
+      logger.severe("Could not get hosts." + e);
```

Comment:
This is a pattern here and throughout this file: You don't want to string-concat the message and 'e', you want to send the full stacktrace through to the logger with the verbose `.log(<level>, <message>, e)` format.